core/src/main/java/jenkins/model/ProjectNamingStrategy.java

```diff
@@ -99,6 +101,11 @@ public DefaultProjectNamingStrategy() {
         @Override
         public void checkName(String origName) throws Failure {
             // default - should just do nothing (this is how Jenkins worked before introducing this ExtensionPoint)
+            if (StringUtils.isNotBlank(DEFAULT_PATTERN) && StringUtils.isNotBlank(origName)) {
```

Comment:
> I am following the way the pattern is used on the rest of the methods. I think this should be fine.

Pattern is time consuming instance. Use it when it really required.

>I think I am going just to do:
Enter space -> get single quote error --> not logical for end users See my comment https://github.com/jenkinsci/jenkins/pull/1943/files#r46538967